### PR TITLE
Update intl doc

### LINKED ISSUE #879: Rendering react-intl strings to static html
The generated html has untranslated keys instead generated ones. 

`<span>footer.subscribe_newsletter</span>`

So if javascript is disabled, there is no text on my website.
It's very problematic for the seo.

### PR DESCRIPTION
**Pull request by xuopled**
So I realize my `redux` solution was not working because the generated `html` files did not contain the translated keys so it was so bad for the seo.

The solution below is better and easier. 

Fix #879

### CODE REVIEW

File: docs/content/docs/usage/intl.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ title: Internationalize your content
 You have multiple solutions available to help you translate your user interface
 and handle multiple locales.
 
-Here we will explain 2 solutions based on
+Here we will the solution based on
```
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: Here we will **explain** _one_ solution based on

File: docs/content/docs/usage/intl.md [hunk 2]
```diff
@@ -40,183 +40,101 @@ Now create differents `.yml` files in a `translations` folder.
...
 
-From now you have 2 variations: with or without redux.
+## Initialize react-intl
 
-Without redux the thing will be pretty simple:
+We will need the `flat` lib to transform our translations files into flatten objects.
+
+```console
+npm i -S flat
+```
```
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: Could you explain use case for flat? I add to open flat doc to understand the idea :)

File: docs/content/docs/usage/intl.md [hunk 3]
```diff
@@ -40,183 +40,101 @@ Now create differents `.yml` files in a `translations` folder.
...
-export default function intlReducer(state = initialState, action) {
-  if (action.type === SET_LOCALE) {
-    return {
-      locale: action.locale,
-      messages: action.messages,
-      formats: action.formats,
-    }
+// We will inject this into IntlProvider
+export function getIntl(location) {
+  const firstURIlevel = location.pathname.replace(/^\//, "").split("/")[0]
```
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: Can you add a comment on this line to explain what it does? Just so people quickly understand this tricky line ^^

File: docs/content/docs/usage/intl.md (hunk 1 above)
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: We also might explain in the top of this doc section that the idea is to have the language in the URL.

File: docs/content/docs/usage/intl.md (hunk 2 above)
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: And here use "install --save" like in the beginning of the doc :)

File: docs/content/docs/usage/intl.md [hunk 4]
```diff
@@ -40,183 +40,101 @@ Now create differents `.yml` files in a `translations` folder.
...
 
-Without redux the thing will be pretty simple:
+We will need the `flat` lib to transform our translations files into flatten objects.
+
+```console
+npm i -S flat
+```
 
 We will assume all your urls will contain the locale key as the first part
+So let's initialize `react-intl` by creating an `src/utils/intl.js` file.
```
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: Can you move this sentence after the block of http://domain?

**Comment by xuopled** [comment]
Thank you, I did modifications.

File: docs/content/docs/usage/intl.md [hunk 5]
```diff
@@ -231,12 +180,13 @@ Your content
...
+import {injectIntl, intlShape} from "react-intl"
+// ...
 class Homepage extends Component {
   render() {
-    const {collection} = this.context
-    const data = enhanceCollection(collection, {
-      filter: {locale: this.props.currentLocale},
+    const data = enhanceCollection(this.context.collection, {
+      // Filter the data by the "locale" front-matter metadata
+      filter: {locale: this.props.intl.locale},
```
Comment (MoOx) [inline docs/content/docs/usage/intl.md]: Pretty sure the filename/url is enough.

File: docs/content/docs/usage/intl.md [hunk 6]
```diff
@@ -40,203 +40,166 @@ Now create differents `.yml` files in a `translations` folder.
...
 
-We will need the `flat` lib to transform our translations files into flatten objects
+Add a metadata `locale` to your `.md` files
 
-```console
-npm i -S flat
+```markdown
+---
+title: Your title
+route: en/your-title
```
Comment (xuopled) [inline docs/content/docs/usage/intl.md]: I remove `locale`metadata and add explaination to use `route` metadata if routes are not good

File: docs/content/docs/usage/intl.md [hunk 7]
```diff
@@ -40,203 +40,166 @@ Now create differents `.yml` files in a `translations` folder.
...
+const messages = {
+  en: flatten(localeEN),
+  fr: flatten(localeFR),
 }
 
-const LOCALES = {
-  "en": compileLocale("en", localeEN),
-  "fr": compileLocale("fr", localeFR),
-}
+export function getLocale(url) {
```
Comment (xuopled) [inline docs/content/docs/usage/intl.md]: Add the `getLocale` from the url function

File: docs/content/docs/usage/intl.md [hunk 8]
```diff
@@ -40,203 +40,166 @@ Now create differents `.yml` files in a `translations` folder.
...
+// ...
 class Homepage extends Component {
   render() {
     const {collection} = this.context
+    const {intl} = this.props
     const data = enhanceCollection(collection, {
-      filter: {locale: this.props.currentLocale},
+      // Filter the data by the "locale" front-matter metadata
+      filter: (c) => c.layout === "Post"
+        && getLocale(c.__url) === intl.locale,
```
Comment (xuopled) [inline docs/content/docs/usage/intl.md]: Use the `getLocale` into the filter function

File: docs/content/docs/usage/intl.md [hunk 9]
```diff
@@ -246,68 +209,52 @@ class Homepage extends Component {
...
+    const postsPage = enhanceCollection(collection, {
+      filter: (c) => (c.layout === "Posts" && getLocale(c.__url) === intl.locale),
+    }).shift()
     return (
       <header>
         <nav>
-          ...
-          <div onClick={() => updateLocale("en")}>{"en"}</div>
-          <div onClick={() => updateLocale("fr")}>{"fr"}</div>
+          <Link className={styles.item} to={postsPage && postsPage.__url}>
```
Comment (xuopled) [inline docs/content/docs/usage/intl.md]: Do you have ideas to prevent to map collection to find our links ?

**Comment by xuopled** [comment]
@MoOx I did some updates

**Comment by xuopled** [comment]
@MoOx I did updates

**Pull request merged**